File: AN_2/SEM_2/MPP/c#/lab3/domain/Cursa.cs
namespace lab3.domain
{
    public class Cursa : Identifiable<int>
    {
        private int idCursa;
        private string nume;
        private int capacitateMinima;
        private int capacitateMaxima;

        public Cursa(string nume, int capacitateMinima, int capacitateMaxima)
        {
            this.nume = nume;
            this.capacitateMinima = capacitateMinima;
            this.capacitateMaxima = capacitateMaxima;
        }

        public Cursa(int idCursa, string nume, int capacitateMinima, int capacitateMaxima)
        {
            this.idCursa = idCursa;
            this.nume = nume;
            this.capacitateMinima = capacitateMinima;
            this.capacitateMaxima = capacitateMaxima;
        }

        public string GetNume() => nume;
        public void SetNume(string nume) => this.nume = nume;

        public int GetCapacitateMinima() => capacitateMinima;
        public void SetCapacitateMinima(int capacitateMinima) => this.capacitateMinima = capacitateMinima;

        public int GetCapacitateMaxima() => capacitateMaxima;
        public void SetCapacitateMaxima(int capacitateMaxima) => this.capacitateMaxima = capacitateMaxima;

        public void SetId(int id) => this.idCursa = id;
        public int GetId() => idCursa;

        public override bool Equals(object? obj)
        {
            if (this == obj) return true;
            if (obj == null || GetType() != obj.GetType()) return false;
            var cursa = (Cursa)obj;
            return idCursa == cursa.idCursa;
        }

        public override int GetHashCode() => idCursa.GetHashCode();

        public override string ToString()
        {
            return $"Cursa{{idCursa={idCursa}, nume='{nume}', capacitateMinima={capacitateMinima}, capacitateMaxima={capacitateMaxima}}}";
        }
    }
}

File: AN_2/SEM_2/MPP/java/lab5/Model/src/main/java/ro/mpp2024/model/Cursa.java
package ro.mpp2024.model;

import java.io.Serializable;
import java.util.Objects;

public class Cursa implements Identifiable<Integer>, Serializable {
    private int idCursa;
    private String nume;
    private int capacitateMinima;
    private int capacitateMaxima;

    public Cursa(String nume, int capacitateMinima, int capacitateMaxima) {
        this.nume = nume;
        this.capacitateMinima = capacitateMinima;
        this.capacitateMaxima = capacitateMaxima;
    }

    public Cursa(int idCursa, String nume, int capacitateMinima, int capacitateMaxima) {
        this.idCursa = idCursa;
        this.nume = nume;
        this.capacitateMinima = capacitateMinima;
        this.capacitateMaxima = capacitateMaxima;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public int getCapacitateMinima() {
        return capacitateMinima;
    }

    public void setCapacitateMinima(int capacitateMinima) {
        this.capacitateMinima = capacitateMinima;
    }

    public int getCapacitateMaxima() {
        return capacitateMaxima;
    }

    public void setCapacitateMaxima(int capacitateMaxima) {
        this.capacitateMaxima = capacitateMaxima;
    }

    @Override
    public void setId(Integer integer) {
        this.idCursa = integer;
    }

    @Override
    public Integer getId() {
        return idCursa;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Cursa cursa = (Cursa) o;
        return idCursa == cursa.idCursa;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idCursa);
    }

    @Override
    public String toString() {
        return "Cursa{" +
                "idCursa=" + idCursa +
                ", nume='" + nume + '\'' +
                ", capacitateMinima=" + capacitateMinima +
                ", capacitateMaxima=" + capacitateMaxima +
                '}';
    }
}
